citylistmetadata/src/parsing.ts

```diff
@@ -34,6 +34,65 @@ export function parseSubjekt(elem: Element): Subjekt | null {
     }
 }
 
+/**
+ * Normalize municipality name by stripping official prefix and fixing case.
+ *
+ * This is just a combination of `stripOfficialNamePrefix` and `normalizeTitleCase`.
+ */
+export function normalizeMunicipalityName(name: string): string {
+    return normalizeTitleCase(stripOfficialNamePrefix(name))
+}
+
+/**
+ * Strip the official name prefix such as “obec” or “městská část” from a municipality name.
+ *
+ * Also strips any whitespace after the prefix, ignores case when matching, does not touch anything else.
+ */
+export function stripOfficialNamePrefix(name: string): string {
+    const prefixes = ['obec', 'město', 'městys', 'městská část', 'městský obvod', 'statutární město']
+    const patterns = prefixes.map(p => RegExp(`^${p}\\s*`, "i"))
+    patterns.forEach(p => name = name.replace(p, ""))
+    return name
+}
+
+export function normalizeTitleCase(name: string): string {
+
+    // Remove spaces around hyphens. According to the rules of Czech typography,
+    // spaces should only be used around a regular dash (–), not hyphen (-). We
+    // are assuming that all hyphens in the source data set are really hyphens,
+    // ie. “Rájec - Jestřebí” really means “Rájec-Jestřebí”.
+    const fixHyphens = (s: string) => s.replace(/\s*-\s*/, "-")
```

Comment:
U víceslovných bychom mohli oddělit pomlčkou místo spojovníkem, vypadá to pak lépe :)

> Spojovník se píše k vyznačení těsného spojení v místních jménech a v názvech správních oblastí, např. Brno-Slatina, Frýdek-Místek, Garmisch-Partenkirchen, Praha-Hlubočepy, Bohumín-sever, Ostrava-město, Praha-západ. Pokud je alespoň jedna ze složek víceslovná, lze z důvodu přehlednosti nahradit spojovník pomlčkou oddělenou mezerami, tedy např. Praha 6 – Ruzyně, Liberec VII – Horní Růžodol, Karlovy Vary – Drahovice, Brno – Královo Pole (viz Pomlčka).
>
> Upozorňujeme, že náhrada spojovníku pomlčkou oddělenou mezerami není vhodná v případech, kdy by mohlo dojít k textové nejednotě (např. v seznamech adres) či k významové nejednoznačnosti (např. při zápisu cest, dopravních spojení, vedení silnic, stezek nebo tras výletů). V takových případech doporučujeme ponechat spojovník, např. železniční trať Ústí nad Labem-Střekov – Děčín hl. n., výluka v úseku Praha-Vysočany – Praha-Horní Počernice – Lysá nad Labem, trasa výletu Praha-Velká Chuchle – Černošice-Mokropsy – Řevnice – Karlštejn apod.

Reply:
Ve zdrojových datech je totální bordel :) Myslel jsem, že ten rozdíl mezi pomlčkou a spojovníkem neumíme algoritmicky vyřešit, ale nejspíš teda umíme, stačí kouknout na počet slov. Mrknu na to, díky!